src/libraries/Microsoft.Extensions.Configuration.Binder/tests/ConfigurationBinderTests.cs

```diff
@@ -812,6 +817,39 @@ public void CanBindValueTypeOptions()
             Assert.Equal("hello world", options.MyString);
         }
 
+        [Fact]
+        public void CanBindByteArray()
+        {
+            var bytes = new byte[] { 1, 2, 3, 4 };
+            var dic = new Dictionary<string, string>
+            {
+                { "MyByteArray", Convert.ToBase64String(bytes) }
+            };
+            var configurationBuilder = new ConfigurationBuilder();
+            configurationBuilder.AddInMemoryCollection(dic);
+            var config = configurationBuilder.Build();
+
+            var options = config.Get<ByteArrayOptions>();
+            Assert.Equal(bytes, options.MyByteArray);
+        }
+
+        public void ExceptionWhenTryingToBindToByteArray()
```

Comment:
Missed to mark as [Fact]?

Reply:
Oops! I amended the commit.

Reply:
As you can guess, I didn't run the test locally. The build process for this repo is _very_ painful, so for simple changes I tend to rely on the CI build.

Reply:
FWIW, using the docs page what I personally usually follow is the instructions from within the [docs/workflow/building/libraries](https://github.com/dotnet/runtime/tree/master/docs/workflow/building/libraries) page. So, I usually just call
```
runtimeRootFolder> git clean -fdx 
runtimeRootFolder> build.cmd clr+libs -rc Release
```
then in this case to build the src project:
```
runtimeRootFolder> dotnet build src\libraries\Microsoft.Extensions.Configuration.Binder\src\ /t:build
```
then to build the test project:
```
runtimeRootFolder> dotnet build src\libraries\Microsoft.Extensions.Configuration.Binder\tests\ /t:test
```

If you want the above procedure to be done even slightly faster and don't care for debugging while you are making your changes, you can just build in release configuration, by appending a ` -c Release` to the end of the `dotnet build` and `build.cmd` commands above.

Reply:
@maryamariyan thanks! I'll try that.
What really bothers me is that the repo is huge and the initial build takes forever (about 2 hours on my Surface Pro, which isn't very powerful and has limited disk space)

Reply:
@thomaslevesque, given that you're working on Microsoft.Extensions.* and those projects target netstandard2.0 and net461, you don't need to build the whole repo and you could just build the source project and the run tests agains net461.

```
dotnet build src\libraries\Microsoft.Extensions.Configuration\src
```

And then:
```
dotnet test src\libraries\Microsoft.Extensions.Configuration\tests -f net461
```

Reply:
@maryamariyan @safern thanks for the suggestions, but nothing seems to work. As a last resort, I eventually tried to build the whole repo as suggested in the docs, but it failed with this error:

```
  CMake Error at CMakeLists.txt:6 (project):
    The CMAKE_C_COMPILER:

      C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.26.28801/bin/Hostx64/x64/cl.exe

    is not a full path to an existing compiler tool.



  CMake Error at CMakeLists.txt:6 (project):
    The CMAKE_CXX_COMPILER:

      C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.26.28801/bin/Hostx64/x64/cl.exe

    is not a full path to an existing compiler tool.
```
I have cl.exe, but it's in 14.27.29110, not 14.26.28801. No idea what I should do to fix it.

I added a test for the null case, as suggested.

Reply:
> error : MembersMustExist : Member 'public T System.Runtime.CompilerServices.Unsafe.Add(T, System.UIntPtr)' does not exist in the reference but it does exist in the implementation.

Ah sorry, I believe you need to build ref assemblies at least. Before the steps I suggested do:

`build.cmd libs.ref` -- then follow the steps I suggested.

@ViktorHofer is this expected?

Reply:
> Ah sorry, I believe you need to build ref assemblies at least. Before the steps I suggested do:
> 
> `build.cmd libs.ref` -- then follow the steps I suggested.

That fails too...

```
C:\Users\thoma\.nuget\packages\microsoft.dotnet.genfacades\5.0.0-beta.20474.4\build\Microsoft.DotNet.GenPartialFacadeSource.targets(30,5): error : There are multiple versions of these assemblies: System.Data.SqlClient.dll. Please provide a single version. [D:\git\dotnet-runtime\src\libraries\shims\generated\netstandard.csproj]
```

Reply:
Could you try doing it on a clean state? i.e, after running `git clean -fdx`?

Reply:
> Could you try doing it on a clean state? i.e, after running `git clean -fdx`?

It worked! Thanks for your help!
This approach is much easier than building the whole repo.

Reply:
@thomaslevesque great to hear that you succeeded. We recently enabled building and testing libraries that don't target .NETCoreApp latest. You shouldn't need to do anything more than a `dotnet test -f <tfm>` to test against .NET Framework and `dotnet build -f <tfm>` to build against .NETStandard. Note that dotnet test already incrementally builds the test and src library as well (like you would expect).

Please let me know if there is something that isn't yet working. Thanks

Reply:
Yeah I think that was a fallout from not having a clean build. I was reacting to the error that suggested a ref was missing/wrong. Thanks @ViktorHofer